charms/worker/k8s/src/charm.py

```diff
@@ -480,7 +480,8 @@ def _assemble_bootstrap_config(self):
         bootstrap_config.extra_sans = self._get_extra_sans()
         cluster_name = self.get_cluster_name()
         node_ips = self._get_node_ips()
-        config.extra_args.craft(self.config, bootstrap_config, cluster_name, node_ips)
+        datastore = self.bootstrap.config.datastore
+        config.extra_args.craft(self.config, bootstrap_config, cluster_name, node_ips, datastore)
```

Comment:
don't we also need this in [`_join_with_token(...)`](https://github.com/canonical/k8s-operator/pull/658/files#diff-5ba1576e3daac57729756610a1f51b9706129e66f250d7a8557ea5666d0c490aR920)